compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1580,11 +1580,15 @@ impl<'tcx> ExistentialProjection<'tcx> {
     ) -> Self {
         // Assert there is a Self.
         projection_predicate.projection_ty.substs.type_at(0);
+        let ty = match projection_predicate.term {
+            Term::Ty(ty) => ty,
+            Term::Const(_c) => unimplemented!(),
```

Comment:
I guess this is where we run into the issue that existential projections also need to be able to handle terms instead of types. Do you plan this for the current PR or as a follow up (you'd need to mark the feature gate as incomplete then)

Reply:
I can stick it in the current PR